### ISSUE TITLE
Examples with switches for all possible compilers?

### ISSUE THREAD

**Issue by tdd11235813** [ComputationalRadiationPhysics/alpaka#502]
The alpaka examples should provide all possible compiler switches, for instance:
```C++
#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
    using Acc = alpaka::acc::AccGpuCudaRt<Dim, Idx>;
    using QueueAcc = alpaka::queue::QueueCudaRtSync;
#elif ALPAKA_ACC_GPU_HIP_ENABLED
    using Acc = alpaka::acc::AccGpuHipRt<Dim, Idx>;
    using QueueAcc = alpaka::queue::QueueHipRtSync;
#else
    using Acc = alpaka::acc::AccCpuSerial<Dim, Idx>;
    using QueueAcc = alpaka::queue::QueueCpuSync;
// ...
// use Acc, QueueAcc, ...
```
That can probably a long thing, but can be a code snippet in a header used by all examples?

At the moment `vectorAdd` uses only the CPU backend. 
For a testing scenario I switched by cmake to CUDA, but I still had to change code in `main.cpp` to test it on CUDA.
So what could be a common approach to have a proper code that tests backends as configured by cmake?

**Comment by tdd11235813** [comment]
agreed. I had assumed that the examples shall work on different accelerators. But there should be kind of an advanced example, which show handling of multiple accelerators. If someone uses alpaka, he likely wants to see that.

**Comment by bernhardmgruber** [comment]
Isn't this already solved by the `<alpaka/example/ExampleDefaultAcc.hpp>`>. See [here](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/example/ExampleDefaultAcc.hpp). And the queue can be created from the accelerator selected by that header.
